src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
```

Comment:
This little "make sure `done` is defined" trick is copied straight from `listener.coffee` from #803. I'll push it down to the middleware executor stage if there's agreement that we should.